src/components/playback/playbackmanager.js

```diff
@@ -1868,38 +1868,43 @@ class PlaybackManager {
                 }, queryOptions));
             } else if (firstItem.Type === 'Series' || firstItem.Type === 'Season') {
                 const apiClient = ServerConnections.getApiClient(firstItem.ServerId);
-                const isSeason = firstItem.Type === 'Season';
+                const startSeasonId = firstItem.Type === 'Season' ? items[options.startIndex || 0].Id : undefined;
 
                 promise = apiClient.getEpisodes(firstItem.SeriesId || firstItem.Id, {
                     IsVirtualUnaired: false,
                     IsMissing: false,
-                    SeasonId: isSeason ? firstItem.Id : undefined,
+                    SeasonId: (startSeasonId && items.length === 1) ? startSeasonId : undefined,
                     SortBy: options.shuffle ? 'Random' : undefined,
                     UserId: apiClient.getCurrentUserId(),
                     Fields: ['Chapters', 'Trickplay']
                 }).then(function (episodesResult) {
-                    const originalResults = episodesResult.Items;
-
-                    let foundItem = false;
-
-                    if (!options.shuffle) {
-                        episodesResult.Items = episodesResult.Items.filter(function (e) {
-                            if (foundItem) {
-                                return true;
+                    if (options.shuffle) {
+                        episodesResult.StartIndex = 0;
+                    } else {
+                        episodesResult.StartIndex = undefined;
+                        let seasonStartIndex;
+                        for (const [index, e] of episodesResult.Items.entries()) {
+                            if (startSeasonId) {
+                                if (e.SeasonId == startSeasonId) {
+                                    if (seasonStartIndex === undefined) {
+                                        seasonStartIndex = index;
+                                    }
+                                } else {
+                                    continue;
+                                }
                             }
-
                             if (!e.UserData.Played) {
-                                foundItem = true;
-                                return true;
+                                episodesResult.StartIndex = index;
+                                break;
                             }
-
-                            return false;
-                        });
+                        }
+                        if (episodesResult.StartIndex === undefined) {
+                            episodesResult.StartIndex = startSeasonId ? seasonStartIndex : 0;
```

Comment:
Is this line testing the correct variable? Should it be testing `seasonStartIndex`?

Reply:
It should work as I had it, but I guess it's possible that `seasonStartIndex` could never get set for some reason. Changed to https://github.com/jellyfin/jellyfin-web/pull/5423/commits/8fcc38a7bcb8b6fd0751c6e6e43abf7027601eac

Reply:
All my tests still work as expected and this was the only line of code that gave me pause.